convert-hf-to-gguf.py

```diff
@@ -2363,6 +2363,14 @@ def set_gguf_parameters(self):
         self.gguf_writer.add_key_length(hparams["head_dim"])
         self.gguf_writer.add_value_length(hparams["head_dim"])
         self.gguf_writer.add_file_type(self.ftype)
+        self.gguf_writer.add_float32(
+            gguf.Keys.LLM.ATTN_LOGIT_SOFTCAPPING.format(arch=self.model_arch),
+            self.hparams["attn_logit_softcapping"]
+        )
+        self.gguf_writer.add_float32(
+            gguf.Keys.LLM.FINAL_LOGIT_SOFTCAPPING.format(arch=self.model_arch),
+            self.hparams["final_logit_softcapping"]
+        )
```

Comment:
I think these need custom `add_` functions like the other parameters.

Reply:
I tried converting the 9b base model and the keys got these names:
```
lama_model_loader: - kv  12:                  27.attn_logit_softcapping f32              = 50.000000
llama_model_loader: - kv  13:                 27.final_logit_softcapping f32              = 30.000000
```
So this isn't being formatted correctly, it is using the arch index instead of the name.

Reply:
It's working now.